v1/storage/inmem/inmem.go

```diff
@@ -349,36 +349,47 @@ func (h *handle) Unregister(_ context.Context, txn storage.Transaction) {
 }
 
 func (db *store) runOnCommitTriggers(ctx context.Context, txn storage.Transaction, event storage.TriggerEvent) {
-	if db.returnASTValuesOnRead && len(db.triggers) > 0 {
-		// FIXME: Not very performant for large data.
+	// While it's unlikely, the API allows one trigger to be configured to want
+	// data conversion, and another that doesn't. So let's handle that properly.
+	var wantsDataConversion bool
+	if db.returnASTValuesOnRead && len(event.Data) > 0 {
+		for _, t := range db.triggers {
+			if !t.SkipDataConversion {
+				wantsDataConversion = true
+				break
+			}
+		}
+	}
 
-		dataEvents := make([]storage.DataEvent, 0, len(event.Data))
+	var converted storage.TriggerEvent
+	if wantsDataConversion {
+		converted = storage.TriggerEvent{
+			Policy:  event.Policy,
+			Data:    make([]storage.DataEvent, 0, len(event.Data)),
+			Context: event.Context,
+		}
 
 		for _, dataEvent := range event.Data {
 			if astData, ok := dataEvent.Data.(ast.Value); ok {
 				jsn, err := ast.ValueToInterface(astData, illegalResolver{})
 				if err != nil {
 					panic(err)
 				}
-				dataEvents = append(dataEvents, storage.DataEvent{
+				converted.Data = append(converted.Data, storage.DataEvent{
 					Path:    dataEvent.Path,
 					Data:    jsn,
 					Removed: dataEvent.Removed,
 				})
-			} else {
-				dataEvents = append(dataEvents, dataEvent)
 			}
 		}
-
-		event = storage.TriggerEvent{
-			Policy:  event.Policy,
-			Data:    dataEvents,
-			Context: event.Context,
-		}
 	}
 
 	for _, t := range db.triggers {
-		t.OnCommit(ctx, txn, event)
+		if wantsDataConversion && !t.SkipDataConversion {
```

Comment:
```suggestion
		if !t.SkipDataConversion {
```

Using `wantsDataConversion` seems unnecessary here, because it will be true if `!t.SkipDataConversion`.